### ISSUE TITLE
Startup crash in linux/libfontconfig issue

### ISSUE THREAD

**Issue by shirishag75** [mars-sim/mars-sim#41]
I got the following crash - 

```
> java -jar mars-sim-main-3.1.0-SNAPSHOT-jar-with-dependencies.jar
Command: java -Djava.util.logging.config.file=logging.properties -cp .:*:jars/* org.mars_sim.msp.javafx.MarsProjectFX -Xms256m -Xmx1024m -new
OUTPUT>23 Mar, 2017 7:26:42 PM(INFO) main.javafx.MarsProjectFX Mars Simulation Project v3.1.0 - Build 3899 - 64-bit 1.8.0_121 - 4 CPU threads
OUTPUT>23 Mar, 2017 7:26:42 PM(INFO) main.javafx.MarsProjectFX Starting a new sim in GUI mode in Linux
OUTPUT>23 Mar, 2017 7:26:43 PM(INFO) ui.javafx.mainmenu.SpinningGlobe JavaFX 3D features supported
OUTPUT>23 Mar, 2017 7:27:19 PM(INFO) core.time.MasterClock CPU Benchmark 1 : 14277 ms
OUTPUT>23 Mar, 2017 7:27:20 PM(INFO) core.structure.Settlement Setting Alpha Base's Objective to Crop Farm
OUTPUT>23 Mar, 2017 7:27:20 PM(INFO) core.structure.Settlement Setting Schiaparelli Point's Objective to Crop Farm
OUTPUT>23 Mar, 2017 7:27:23 PM(SEVERE) core.time.UpTimer Too long since last time pulse, clearing update time
OUTPUT>#
OUTPUT># A fatal error has been detected by the Java Runtime Environment:
OUTPUT>#
OUTPUT>#  SIGSEGV (0xb) at pc=0x00007f38c055d820, pid=25165, tid=0x00007f3860104700
OUTPUT>#
OUTPUT># JRE version: OpenJDK Runtime Environment (8.0_121-b13) (build 1.8.0_121-8u121-b13-4-b13)
OUTPUT># Java VM: OpenJDK 64-Bit Server VM (25.121-b13 mixed mode linux-amd64 compressed oops)
OUTPUT># Problematic frame:
OUTPUT># C  [libfontconfig.so.1+0x19820]
OUTPUT>#
OUTPUT># Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
OUTPUT>#
OUTPUT># An error report file with more information is saved as:
OUTPUT># /media/shirish/Seagate-Slim-Backup/linux-games/mars-sim.git/trunk/mars-sim-main/target/hs_err_pid25165.log
OUTPUT>[thread 139880135108352 also had an error]
OUTPUT>#
OUTPUT># If you would like to submit a bug report, please visit:
OUTPUT>#   http://bugreport.java.com/bugreport/crash.jsp
OUTPUT># The crash happened outside the Java Virtual Machine in native code.
OUTPUT># See problematic frame for where to report the bug.
OUTPUT>#
```
As 

[hs_err_pid25165.txt](https://github.com/mars-sim/mars-sim/files/864891/hs_err_pid25165.txt)

**Comment by shirishag75** [comment]
I again did it and got the same crash. This tme the log has more details, 
[hs_err_pid30286.txt](https://github.com/mars-sim/mars-sim/files/864985/hs_err_pid30286.txt)

maybe this one gives you more info.

**Comment by mokun** [comment]
Do you have more than one instance of mars-sim running ? If you terminate other old mars-sim running, it may lessen the chance of the crash.

I got Ubuntu setup. After that I did see similar crash from time to time with OpenJDK.

The thing is that I never see this in window, 

It said

**Comment by shirishag75** [comment]
> Do you have more than one instance of mars-sim running ? If you terminate other old mars-sim running, it may lessen the chance of the crash.

No, there is no other instance of mars-sim running. What I did was simply run the game, it asked me of the two scenarios, which one to take and I took the one which had more men, a somewhat more established colony and after that it crashed. 

I am assuming the same as you, that it might have been a font configuration issue. Lemme know when I should try again.

**Comment by mokun** [comment]
how often does it crash in your machine ?

To me, it only happens may be once every 4 times I run eclipse launcher.

I stick with windows most of the time and it never happens. So, it will have to do with how JVM handles the font in linux.

**Comment by shirishag75** [comment]
everytime. I haven't been able to look into anything after starting the settlement, it crashes right after I choose which settlement and I have chosen both, one at one time and the other at other time.

**Comment by mokun** [comment]
You are running the compiled jar and it looks like it wasn't compiled correctly.

Can you just run the MarsProjectFX's main class and see what happen ?

To me, I run maven only when I make certain critical changes and would need to run the test cases to verify the changes would not break anything. 

So after I made a change, 9 out of 10 times, I would simply run the Eclipse launcher that will append the needed arguments to MarsProjectFX's main class as I run it.

In other words, you don't need to run maven all the time. You need it only when you want to run test cases and/or package into the compiled jar.

**Comment by shirishag75** [comment]
@mokun how do I do that ?  Need the exact command, java isn't my cup of tea (unfortunately) .

**Comment by mokun** [comment]
I just remove the need of external fonts in r3902[](https://github.com/mars-sim/mars-sim/commit/325a96470b20055911a6657180107a3d211d0ae5). The startup crash no longer happens in ubuntu.

Can you test it in debian ?

@larsnaesbye, do you know the exact java commands for compiling/building/running the .class files for mars-sim in command console. I use Eclipse launcher and don't know right off hand what they are. I know Scott use command line as well but we can't reach him.

**Comment by mokun** [comment]
> has to do with how JVM handles external fonts in linux via libfontconfig.

Using Oracle's Java 8 (instead of OpenJDK with OpenJFX) seems to eliminate the font issue. Someone will need to check it.

Recommend closing the issue unless it resurfaces again.
